modules/tinymce/src/core/main/ts/content/SetContentImpl.ts

```diff
@@ -35,9 +35,11 @@ const moveSelection = (editor: Editor): void => {
   }
 };
 
-const setEditorHtml = (editor: Editor, html: string): void => {
+const setEditorHtml = (editor: Editor, html: string, noSelection: boolean = false): void => {
   editor.dom.setHTML(editor.getBody(), html);
-  moveSelection(editor);
+  if (!noSelection) {
```

Comment:
This will also need changing if we do the above, e.g.:
```suggestion
  if (noSelection !== true) {
```